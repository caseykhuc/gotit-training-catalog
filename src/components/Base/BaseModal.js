import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import { connect } from 'react-redux';
import PropTypes from 'prop-types'
import { hideModal } from '../../actions/modal';

export const BaseModal = ({
  title, hideModal, onAccept, children,
}) => (
    <Modal show onHide={hideModal} centered>
      <Modal.Header closeButton>
        <Modal.Title>{title}</Modal.Title>
      </Modal.Header>
      <Modal.Body>{children}</Modal.Body>
      <Modal.Footer>
        <Button variant="secondary" onClick={hideModal}>
          Close
      </Button>
        <Button variant="primary" onClick={onAccept}>
          Accept
      </Button>
      </Modal.Footer>
    </Modal>
  )

BaseModal.propTypes = {
  title: PropTypes.string.isRequired,
  hideModal: PropTypes.func.isRequired,
  onAccept: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
}

export default connect(null, { hideModal })(BaseModal);
