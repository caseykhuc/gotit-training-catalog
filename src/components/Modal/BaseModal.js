import React from 'react';
import { Modal, Button } from 'react-bootstrap';
import { hideModal } from '../../actions';
import { connect } from 'react-redux';

class BaseModal extends React.Component {
  render() {
    return (
      <Modal.Dialog>
        <Modal.Header closeButton onClick={this.props.hideModal}>
          <Modal.Title>Modal title</Modal.Title>
        </Modal.Header>

        <Modal.Body>
          <p>Modal body text goes here.</p>
        </Modal.Body>

        <Modal.Footer>
          <Button variant='secondary'>Close</Button>
          <Button variant='primary'>Save changes</Button>
        </Modal.Footer>
      </Modal.Dialog>
    );
  }
}

export default connect(null, { hideModal })(BaseModal);
