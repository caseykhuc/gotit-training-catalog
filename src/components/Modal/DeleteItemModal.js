import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import BaseModal from 'components/Base/BaseModal';
import confirmMessage from 'utils/confirmMessage';
import { deleteItem } from 'actions/item';

export const DeleteItemModal = ({ categoryId, itemId, deleteItem }) => (
  <BaseModal
    title="DELETE CONFIRM"
    onAccept={() => deleteItem(categoryId, itemId)}
  >
    <p>{confirmMessage.confirmDelete}</p>
  </BaseModal>
)

DeleteItemModal.propTypes = {
  categoryId: PropTypes.number.isRequired,
  itemId: PropTypes.number.isRequired,
  deleteItem: PropTypes.func.isRequired,
}

export default connect(null, { deleteItem })(DeleteItemModal)
