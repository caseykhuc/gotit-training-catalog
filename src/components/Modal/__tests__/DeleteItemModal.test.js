import React from 'react';
import { shallow } from 'enzyme';
import { DeleteItemModal } from '../DeleteItemModal';
import BaseModal from '../../Base/BaseModal';

describe('components/Modal/DeleteItemModal', () => {
  let props;
  let wrapper;
  let baseModal;

  const update = () => {
    wrapper.update();
    baseModal = wrapper.find(BaseModal);
  };
  const setup = () => {
    wrapper = shallow(<DeleteItemModal {...props} />);
    update();
  };
  beforeEach(() => {
    props = {
      categoryId: 2,
      itemId: 1,
      deleteItemAndRefetch: jest.fn(),
    }
  });
  it('should render correctly', () => {
    setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('should invoke action when BaseModal invoked onAccept ', () => {
    setup();
    baseModal.props().onAccept();
    expect(props.deleteItemAndRefetch).toHaveBeenCalledWith(props.categoryId, props.itemId);
  });
})
