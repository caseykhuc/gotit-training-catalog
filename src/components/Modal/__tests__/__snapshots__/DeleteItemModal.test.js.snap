// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Common/DeleteItemModal should render correctly 1`] = `ShallowWrapper {}`;
